extension/solarwindsextension/internal/heartbeat.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 SolarWinds Worldwide, LLC. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	"errors"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+	"sync"
+	"time"
+)
+
+type Heartbeat struct {
+	logger *zap.Logger
+
+	cancel           context.CancelFunc
+	startShutdownMtx sync.Mutex
+
+	metric        *UptimeMetric
+	exporter      *Exporter
+	collectorName string
+}
+
+var alreadyRunningError = errors.New("heartbeat already running")
```

Comment:
Minor: I'd follow the naming convention for sentinel errors from the [stdlib](https://pkg.go.dev/os/exec#pkg-variables), i.e. `ErrAlreadyRunning` and make it public as the error is returned from a public function.

Reply:
Yes, done in https://github.com/solarwinds/solarwinds-otel-collector/pull/4/commits/413b882ddadc700ba7604ad8cbd28523c0fc742a